sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -117,12 +117,25 @@ private[sql] class HiveSessionCatalog(
     try {
       lookupFunction0(name, children)
     } catch {
-      case NonFatal(_) =>
```

Comment:
Is it bad to just add some checks here like this?
```
      case NonFatal(_) if children.exists(_.dataType.isInstanceOf[DecimalType]) =>
```

Reply:
This way looks better. But we still need to append another `case NonFatal(e)` below to make sure that exception will be threw finally for the function without any `DecimalType` children.

Reply:
> But we still need to append another case NonFatal(e) below to make sure that exception will be threw finally for the function without any DecimalType children.

Could you please elaborate? If I'm not missing anything, the exception doesn't match any cases, exception will be propagated to caller.

Reply:
You'r right, maybe. But we need to match non fatal error if we want to be consistent with original behavior.

Reply:
At least this patch will throw origin exception if there's no matching DecimalType which will be same as what @maropu is suggested, and if we would like to avoid calling lookup0 if it doesn't need to, throwing origin exception would be enough which can be done with not matching the case.

Reply:
@maropu at first i also write like this... but I also want to avoid tranverse the expressions of children twice (by `.exists` and `.map`)

Reply:
That's fine. Let's just go simple. The original fix itself is a partial band-aid fix.

Reply:
Ah for clarification, let's go for Takeshi's suggestion :).